### PR TITLE
fix session ID references when using mixed repeat/normal subcases

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?164597

most relevant for @snopoke but @dannyroberts @benrudolph or @nickpell are encouraged to review.

### CODE REVIEW

File: corehq/apps/app_manager/suite_xml.py [hunk 1]
```diff
@@ -1504,14 +1504,16 @@ def get_new_case_id_datums_meta(self, form):
...
+                for i, subcase in enumerate(actions['subcases']):
+                    # don't put this in the loop to be consistent with the form's indexing
+                    # see XForm.create_casexml_2
+                    if not subcase.repeat_context:
+                        datums.append({
+                            'datum': SessionDatum(id=form.session_var_for_action('subcase', i), function='uuid()'),
+                            'case_type': subcase.case_type,
+                            'requires_selection': False,
+                            'action': subcase
+                        })
```
Comment (czue) [inline corehq/apps/app_manager/suite_xml.py]: this is the only change. basically when the form generates it session IDs it does so indexing over the entire list of subcase actions, but when the suite does it it was pruning them off first. as a result if you had repeats mixed with regular cases the indices would be off-by-one (or however many repeats you had)

**Comment by czue** [comment]
guessing this has been around since https://github.com/dimagi/commcare-hq/pull/5851/ though is likely a very small set of apps that are affected. not sure how much effort we want to put into finding and fixing bad builds

File: corehq/apps/app_manager/suite_xml.py [hunk 2]
```diff
@@ -1504,14 +1504,16 @@ def get_new_case_id_datums_meta(self, form):
                 })
 
             if 'subcases' in actions:
-                non_repeat_actions = [a for a in actions['subcases'] if not a.repeat_context]
```
Comment (dannyroberts) [inline corehq/apps/app_manager/suite_xml.py]: Oh wow, that is super sneaky!

**Comment by dannyroberts** [comment]
Good to go pending build! (Using w=1 made it really easy to tell whether there were any changes other than moving the filter into the loop: https://github.com/dimagi/commcare-hq/pull/6236/files?w=1#diff-27910c0ed3e4d6d075211ffc4db6cab4L1507. There weren't!)

File: corehq/apps/app_manager/suite_xml.py (hunk 1 above)
Comment (snopoke) [inline corehq/apps/app_manager/suite_xml.py]: Nice find Cory.

**Comment by czue** [comment]
:white_check_mark:

**Comment by benrudolph** [comment]
wow nice, this would've took me a while to get. thanks for crushing it

**Pull request merged**